nixos/modules/services/mail/goeland.nix

```diff
@@ -0,0 +1,58 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.goeland;
+  tomlFormat = pkgs.formats.toml { };
+in
+{
+  options.services.goeland = {
+    enable = mkEnableOption (mdDoc "goeland");
+
+    settings = mkOption {
+      description = mdDoc ''
+        Configuration of goeland.
+        See the [example config file](https://github.com/slurdge/goeland/blob/master/cmd/asset/config.default.toml) for the available options.
+      '';
+      default = { };
+      type = types.submodule {
+        freeformType = tomlFormat.type;
+      };
+    };
+    schedule = mkOption {
+      type = types.str;
+      default = "12h";
+      example = "Mon, 00:00:00";
+      description = mdDoc "How often to run goeland, in systemd time format";
+    };
+    databaseDirectory = mkOption {
+      type = types.path;
+      default = "/var/lib/goeland";
+      description = mdDoc "Directory where the goeland database will reside if using the `unseen` filter";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.tmpfiles.rules = [ "d ${cfg.databaseDirectory} 0750 goeland goeland -" ];
+
+    services.goeland.settings.database = "${cfg.databaseDirectory}/goeland.db";
+
+    systemd.services.goeland = {
```

Comment:
I think you have to explicitly ask systemd for a `StateDirectory=goeland` for it to manage the /var/lib/goeland permissions.

Reply:
Yes, I hadn't pushed any changes yet, I was waiting to see what you would answer to my question.
Now it's done.